janet-http/http-service/src/main/java/io/techery/janet/HttpActionService.java

```diff
@@ -85,48 +87,79 @@ public HttpActionService(String baseUrl, HttpClient client, Converter converter)
     @Override protected <A> void sendInternal(ActionHolder<A> holder) throws HttpServiceException {
         callback.onStart(holder);
         A action = holder.action();
-        runningActions.add(action);
+        putRunningAction(action);
         final ActionHelper<A> helper = getActionHelper(action.getClass());
         if (helper == null) {
             throw new JanetInternalException("Something was happened with code generator. Check dependence of janet-http-compiler");
```

Comment:
no. we need to know when action started but doesn't have an request yet.
Also we don't know how long will the method `helper.fillRequest` be running.